Assets/HoloToolkit/Utilities/Editor/ProjectSettingsWindow.cs

```diff
@@ -0,0 +1,274 @@
+								foundBuildTargetVRSettings = true;
+								foundBuildTargetMetro = true;
+								foundBuildTargetEnabled = true;
+								foundDevices = true;
+								foundHoloLens = true;
+							}
+							else
+							{
+								// The target VR settngs were found but the others
+								// still need to be searched for.
+								foundBuildTargetVRSettings = true;
+							}
+						}
+
+					}
+
+					// Look for the build target for Metro
+					else if (!foundBuildTargetMetro)
+					{
+						if (line.Contains("m_BuildTarget: Metro"))
+						{
+							foundBuildTargetMetro = true;
+						}
+
+					}
+
+					else if (!foundBuildTargetEnabled)
+					{
+						if (line.Contains("m_Enabled"))
+						{
+							line = "    m_Enabled: 1";
+							foundBuildTargetEnabled = true;
+						}
+					}
+
+					// Look for the enabled Devices list
+					else if (!foundDevices)
+					{
+						if (line.Contains("m_Devices:"))
+						{
+							// Clear the empty array symbols if any
+							line = line.Replace(" []", "");
+							foundDevices = true;
+						}
+
+					}
+
+					// Once we've found the list look for HoloLens or the next non element
+					else if (!foundHoloLens)
+					{
+						// If this isn't an element in the device list
+						if (!line.Contains("-"))
+						{
+							// add the hololens element, and mark it found
+							builder.Append("    - HoloLens\n");
+							foundHoloLens = true;
+						}
+
+						// Otherwise test if this is the hololens device
+						else if (line.Contains("HoloLens"))
+						{
+							foundHoloLens = true;
+						}
+					}
+
+					builder.Append(line);
+
+					// Write out a \n for all but the last line
+					// NOTE: Specifically preserving unix line endings by avoiding StringBuilder.AppendLine
+					if (i != lines.Length - 1)
+						builder.Append('\n');
+				}
+
+				// Capture the final string
+				settings = builder.ToString();
+
+				File.WriteAllText(settingsPath, settings);
+			}
+			catch (Exception e)
+			{
+				Debug.LogException(e);
+			}
+		}
+
+		/// <summary>
+		/// Modifies the WSA default quality setting to the fastest
+		/// </summary>
+		private void SetFastestDefaultQuality()
+		{
+			try
+			{
+				// Find the WSA element under the platform quality list and replace it's value with 0
+				string settingsPath = "ProjectSettings/QualitySettings.asset";
+				string matchPattern = @"(m_PerPlatformDefaultQuality.*Windows Store Apps:) (\d+)";
+				string replacePattern = @"$1 0";
+
+				string settings = File.ReadAllText(settingsPath);
+				settings = Regex.Replace(settings, matchPattern, replacePattern, RegexOptions.Singleline);
+
+				File.WriteAllText(settingsPath, settings);
+			}
+			catch (Exception e)
+			{
+				Debug.LogException(e);
+			}
+		}
+
+		// Overrides
+		protected override void ApplySettings()
+		{
+			// See the blow notes for why text asset serialization is required
+			if (EditorSettings.serializationMode != SerializationMode.ForceText)
```

Comment:
I'm very much in agreement with you about leveraging text serialization and visible metadata files when sharing Unity projects with a team. However, the only reason text-based serialization is enabled in this context is because some of the settings we want to change (namely enabling VR for WSA and adding HoloLens as a VR device) are not exposed via APIs. Therefore we have to change them in the project file, which is obviously much easier to do if the project is text instead of binary.

We could add an option here to turn on visible metadata as well, but that's not really a "HoloLens setting". It's more of a workflow setting so I don't feel it belongs in this window. I'm open to hearing feedback from others on this though. The way I wrote these windows it's extremely easy to add new options.